src/main/java/org/jabref/gui/groups/GroupsPreferences.java

```diff
@@ -1,40 +1,64 @@
 package org.jabref.gui.groups;
 
+import java.util.EnumSet;
+
 import javafx.beans.property.BooleanProperty;
 import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SetProperty;
 import javafx.beans.property.SimpleBooleanProperty;
 import javafx.beans.property.SimpleObjectProperty;
+import javafx.beans.property.SimpleSetProperty;
+import javafx.collections.FXCollections;
 
 import org.jabref.model.groups.GroupHierarchyType;
 
 public class GroupsPreferences {
 
-    private final ObjectProperty<GroupViewMode> groupViewMode;
+    private final SetProperty<GroupViewMode> groupViewMode;
     private final BooleanProperty shouldAutoAssignGroup;
     private final BooleanProperty shouldDisplayGroupCount;
     private final ObjectProperty<GroupHierarchyType> defaultHierarchicalContext;
 
-    public GroupsPreferences(GroupViewMode groupViewMode,
+    public GroupsPreferences(boolean viewModeIntersection,
+                             boolean viewModeFilter,
+                             boolean viewModeInvert,
```

Comment:
Please create another constructor accepting an EnumSet instead of all the trues and falses.

Reply:
Done in:[`225af78` (#11510)](https://github.com/JabRef/jabref/pull/11510/commits/225af7851432f8869e9cd4e75c92bf86c5187afe)